src/chexus/tree.py

```diff
@@ -16,6 +18,20 @@ class Dataset:
     parent: Group
     attrs: dict[str, Any] = field(default_factory=dict)
     value: Any | None = None
+    dataset: h5py.Dataset | None = None
+
+    @property
+    def value(self) -> Any | None:  # noqa: F811
+        if self.dataset is not None:
+            try:
+                return self.dataset.asstr()[()]
+            except TypeError:
+                return self.dataset[()]
```

Comment:
Should this set `self._value`, and return it if not `None`?

Reply:
If we did that we would store the data in memory and risk OOM if the file is large.

The current solution might read the same data several times, but there's less risk for OOM. Basically trading performance for stability.

If we find this is a performance issue we can always revisit it later and optimize, for example by doing like you suggested.

Reply:
Makes sense. Can you put that in a code comment?